doc/manual/interfaces.rst

```diff
@@ -0,0 +1,223 @@
+.. _man-interfaces:
+
+************
+ Interfaces
+************
+
+A lot of the power and extensibility in Julia comes from a collection of informal interfaces.  By extending few specific methods to work for a custom type, objects of that type not only receive those functionalities, but they are also able to be used in other methods that are written to generically build upon those behaviors.
+
+Iteration
+---------
+
+================================= ======================== ===========================================
+Required methods                                           Brief description
+================================= ======================== ===========================================
+:func:`start(iter) <start>`                                Returns the initial iteration state
+:func:`next(iter, state) <next>`                           Returns the current item and the next state
+:func:`done(iter, state) <done>`                           Tests if there are any items remaining
+**Important optional methods**    **Default definition**   **Brief description**
+:func:`eltype(IterType) <eltype>` ``Any``                  The container's element type
```

Comment:
How about "the type of each item returned by `next`"?  Describing it as the element type is a bit circular.